[python] add context manager support for xbmcvfs.File

Rechi opened the pull request:
## Description
This makes xbmcvfs.File usable in with statements.

```
with xbmcvfs.File(file) as file:
  pass
```

## Motivation and Context
make xbmcvfs.File usage more user-friendly

## How Has This Been Tested?
```
import os
import xbmc
import xbmcaddon
import xbmcvfs

profile = xbmcaddon.Addon().getAddonInfo("profile")
filepath = os.path.join(profile, "test.txt")

xbmcvfs.mkdir(profile)

with xbmcvfs.File(filepath, "w") as file:
    file.write("test")

with xbmcvfs.File(filepath) as file:
    xbmc.log(file.read())
```

## Types of change
- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:
- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [x] All new and existing tests passed

enen92 commented on xbmc/interfaces/legacy/File.h:
```diff
@@ -38,6 +38,7 @@ namespace XBMCAddon
     ///
     ///
     ///--------------------------------------------------------------------------
+    /// @python_v19 Added context manager support
```
maybe update the example as well?

Rechi replied:
@enen92 I added new examples, so one still has the examples for older Kodi versions.

Rechi committed:
[python] add context manager support for xbmcvfs.File

This makes xbmcvfs.File usable in with statements.

with xbmcvfs.File(file) as file:
  pass

enen92 commented:
thanks, looks good to me. There are a few obvious candidates for something similar, for example some gui dialogs (e.g. dialogprogress). Maybe now that I know how easy this is I can replicate later to other sections.
